### ISSUE TITLE
Sphinx 3.1.0 breaks backward compatibility for autofunction

### ISSUE THREAD

**Issue by leofang** [sphinx-doc/sphinx#7809]
In https://github.com/cupy/cupy/pull/3416 we had to temporarily regress to 3.0.4, because we encountered this error in the CI:
```
335 Warning, treated as error:
336 /home/travis/build/cupy/cupy/docs/source/reference/generated/cupy.fft.fft.rst:6:Error in "autofunction" directive:
337 1 argument(s) required, 0 supplied.
338 
339 .. autofunction::
340 Makefile:56: recipe for target 'html' failed
341 make: *** [html] Error 2
342 The command "SPHINXOPTS=-W make html" exited with 2.
```
See the full Travis log here: https://travis-ci.org/github/cupy/cupy/builds/696302151.

cc: @emcastillo

**Comment by tk0miya** [comment]
I think #7815 will also resolve this case. Could you try it please?

**Comment by tk0miya** [comment]
I confirmed the HEAD of 3.1.x branch works fine for copy. So I'm closing this now. It will be released as 3.1.1 soon.
```
FROM python:3.8-slim

RUN apt update; apt install -y git make build-essential vim
RUN git clone https://github.com/cupy/cupy
WORKDIR /cupy
ENV CUPY_CI=travis
ENV READTHEDOCS=True
RUN pip install cython
RUN pip install -e .[travis]
WORKDIR /cupy/docs
RUN pip install -r requirements.txt
RUN pip install git+https://github.com/sphinx-doc/sphinx@3.1.x
RUN make html
```

**Comment by leofang** [comment]
Hi @tk0miya Thanks for quick action and test! I can confirm that when CuPy's Travis picks up 3.1.1 it does not complain: https://travis-ci.org/github/cupy/cupy/builds/698120966. Will report back if we encounter any issue in a full CI build.
